src/amplitude_experiment/assignment/assignment_service.py

```diff
@@ -11,23 +11,35 @@
 def to_event(assignment: Assignment) -> BaseEvent:
     event = BaseEvent(event_type='[Experiment] Assignment', user_id=assignment.user.user_id,
                       device_id=assignment.user.device_id, event_properties={}, user_properties={})
-    for key in sorted(assignment.results):
-        event.event_properties[key + '.variant'] = assignment.results[key].variant['key']
-
     set_props = {}
     unset_props = {}
 
-    for key in sorted(assignment.results):
-        if assignment.results[key].type == FLAG_TYPE_MUTUAL_EXCLUSION_GROUP:
+    for flag_key in sorted(assignment.results):
+        variant = assignment.results[flag_key]
+        if variant.key is None:
+            continue
+        # Get variant metadata
+        version: int = variant.metadata.get('flagVersion') if variant.metadata is not None else None
+        segment_name: str = variant.metadata.get('segmentName') if variant.metadata is not None else None
+        flag_type: str = variant.metadata.get('flagType') if variant.metadata is not None else None
+        default: bool = False
+        if variant.metadata is not None and variant.metadata.get('default') is not None:
+            default = variant.metadata.get('default')
+        # Set event properties
+        event.event_properties[flag_key + '.variant'] = variant.key
+        if version is not None and segment_name is not None:
+            event.event_properties[flag_key + '.details'] = f"v{version} rule:{segment_name}"
```

Comment:
@tyiuhc can you update your php implementation to add the details event property using the flagVersion, segmentName metadata. Also use the flagType and default metadata for determining the user property setting/unsetting behavior as per here.

Reply:
Will do 👍🏻